app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/models/db/EpisodeDaoTest.kt

```diff
@@ -160,4 +163,40 @@ class EpisodeDaoTest {
         )
         assertEquals(expected, statistics)
     }
+
+    @Test
+    fun getFilteredPlaybackHistoryResultForMatchedEpisodeTitle() = runTest {
+        val query = "%test%"
+        val episodes = listOf(PodcastEpisode(uuid = "1", title = "Test Episode", podcastUuid = "podcast_uuid", publishedDate = Date(), lastPlaybackInteraction = 1000))
+        val podcast = Podcast(uuid = "podcast_uuid")
+        episodeDao.insertAll(episodes)
+        podcastDao.insert(podcast)
+
+        val result = episodeDao.filteredPlaybackHistoryFlow(query).first()
+        assertEquals(episodes, result)
+    }
+
+    @Test
+    fun getFilteredPlaybackHistoryResultForMatchedPodcastTitle() = runTest {
+        val query = "%test%"
+        val episodes = listOf(PodcastEpisode(uuid = "1", title = "Episode", podcastUuid = "podcast_uuid", publishedDate = Date(), lastPlaybackInteraction = 1000))
+        val podcast = Podcast(uuid = "podcast_uuid", title = "Test Podcast")
+        episodeDao.insertAll(episodes)
+        podcastDao.insert(podcast)
+
+        val result = episodeDao.filteredPlaybackHistoryFlow(query).first()
+        assertEquals(episodes, result)
+    }
+
+    @Test
+    fun getOrderedFilteredPlaybackHistoryResults() = runTest {
+        val query = "%test%"
+        val episode1 = PodcastEpisode(uuid = "1", title = "Test Episode 1", publishedDate = Date(), lastPlaybackInteraction = 1000)
+        val episode2 = PodcastEpisode(uuid = "2", title = "Test Episode 2", publishedDate = Date(), lastPlaybackInteraction = 2000)
+        val episodes = listOf(episode2, episode1)
+        episodeDao.insertAll(episodes)
+
+        val result = episodeDao.filteredPlaybackHistoryFlow(query).first()
+        assertEquals(episodes, result)
```

Comment:
@ashiagr Do you think it would make sense to add a test for no matched results?

Reply:
Added in a657423